app/components/candidate_interface/rejection_reasons_component.rb

```diff
@@ -0,0 +1,33 @@
+module CandidateInterface
+  class RejectionReasonsComponent < ViewComponent::Base
+    include ViewHelper
+    validates :application_form, presence: true
+
+    def initialize(application_form:)
+      @application_form = application_form
+    end
+
+    def application_choice_rows(application_choice)
+      [
+        course_details_row(application_choice),
+        rejection_reasons_row(application_choice),
+      ].compact
+    end
+
+  private
+
+    def course_details_row(application_choice)
+      {
+        key: 'Course',
+        value: govuk_link_to("#{application_choice.offered_course.name} (#{application_choice.offered_course.code}) <br>".html_safe, application_choice.offered_course.find_url, target: '_blank', rel: 'noopener') + application_choice.course.description
```

Comment:
```suggestion
        value: govuk_link_to("#{application_choice.offered_course.name_and_code} <br>".html_safe, application_choice.offered_course.find_url, target: '_blank', rel: 'noopener') + application_choice.course.description
```

I'd also prefer not to use `html_safe` unless it's absolutely necessary. Can you use [safe_join](https://apidock.com/rails/ActionView/Helpers/OutputSafetyHelper/safe_join) or something similar?

Reply:
have addressed in the other PR